src/main/java/maybeweijun/exception/maybeweijunException.java

```diff
@@ -0,0 +1,164 @@
+package maybeweijun.exception;
+
+/**
+ * Root exception type for the application. Specific error cases are represented
+ * as nested static subclasses to provide semantic meaning and user-friendly messages.
+ */
+public class maybeweijunException extends Exception {
+    public maybeweijunException() {
+        super();
+    }
+
+    public maybeweijunException(String message) {
+        super(message);
+    }
```

Comment:
Perhaps the class name should be in PascalCase rather than CamelCase considering the Java coding standard?
```suggestion
public class MaybeweijunException extends Exception {
    public MaybeweijunException() {
        super();
    }

    public MaybeweijunException(String message) {
        super(message);
    }
```